File: demo/java-demo/src/main/java/com/alibaba/demo/basic/DemoMock.java
package com.alibaba.demo.basic;

import com.alibaba.demo.basic.model.mock.BlackBox;

import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.stream.Collectors;

/**
 * 演示基本的Mock功能
 * Demonstrate basic mock functionality
 */
public class DemoMock {

    /**
     * method with new operation
     */
    public String newFunc() {
        BlackBox component = new BlackBox("something");
        return component.get();
    }

    /**
     * method with member method invoke
     */
    public String outerFunc(String s) throws Exception {
        return "{ \"res\": \"" + innerFunc(s) + staticFunc() + "\"}";
    }

    /**
     * method with common method invoke
     */
    public String commonFunc() {
        return "anything".trim() + "__" + "anything".substring(1, 2) + "__" + "abc".startsWith("ab");
    }

    /**
     * method with static method invoke
     */
    public BlackBox getBox() {
        return BlackBox.secretBox();
    }

    /**
     * two methods invoke same private method
     */
    public String callerOne() {
        return callFromDifferentMethod();
    }

    public String callerTwo() {
        return callFromDifferentMethod();
    }

    private static String staticFunc() {
        return "_STATIC_TAIL";
    }

    private String innerFunc(String s) throws Exception {
        return Files.readAllLines(Paths.get("/a-not-exist-file")).stream().collect(Collectors.joining());
    }

    private String callFromDifferentMethod() {
        return "realOne";
    }

}

File: demo/kotlin-demo/src/main/kotlin/com/alibaba/demo/basic/DemoMock.kt
package com.alibaba.demo.basic

import com.alibaba.demo.basic.model.mock.BlackBox
import com.alibaba.demo.basic.model.mock.ColorBox
import java.nio.file.Files
import java.nio.file.Paths

/**
 * 演示基本的Mock功能
 * Demonstrate basic mock functionality
 */
class DemoMock {

    /**
     * method with new operation
     */
    fun newFunc(): String? {
        return BlackBox("something").get()
    }

    /**
     * method with member method invoke
     */
    fun outerFunc(s: String): String {
        return "{ \"res\": \"" + innerFunc(s) + staticFunc() + "\"}"
    }

    /**
     * method with common method invoke
     */
    fun commonFunc(): String {
        val box = BlackBox("anything")
        return box.trim() + "__" + box.substring(1, 2) + "__" + box.startsWith("any")
    }

    /**
     * method with static method invoke
     */
    fun getBox(): BlackBox {
        return ColorBox.createBox("Red", BlackBox.secretBox())
    }

    /**
     * two methods invoke same private method
     */
    fun callerOne(): String {
        return callFromDifferentMethod()
    }

    fun callerTwo(): String {
        return callFromDifferentMethod()
    }

    private fun innerFunc(s: String): String {
        return Files.readAllLines(Paths.get("/a-not-exist-file")).joinToString()
    }

    private fun callFromDifferentMethod() = "realOne"

    companion object {
        private fun staticFunc(): String {
            return "_STATIC_TAIL"
        }

//        fun callStaticFunc(): String {
//            return "CALL${staticFunc()}"
//        }
    }
}
